samples/snippets/src/test/java/com/example/spanner/DatabaseRolesIT.java

```diff
@@ -0,0 +1,132 @@
+    final String database = idGenerator.generateDatabaseId();
+    databaseAdminClient
+        .createDatabase(
+            instanceId,
+            database,
+            ImmutableList.of(
+                "CREATE TABLE Singers ("
+                    + "  SingerId   INT64 NOT NULL,"
+                    + "  FirstName  STRING(1024),"
+                    + "  LastName   STRING(1024),"
+                    + "  SingerInfo BYTES(MAX),"
+                    + "  FullName STRING(2048) AS "
+                    + "  (ARRAY_TO_STRING([FirstName, LastName], \" \")) STORED"
+                    + ") PRIMARY KEY (SingerId)",
+                "CREATE TABLE Albums ("
+                    + "  SingerId     INT64 NOT NULL,"
+                    + "  AlbumId      INT64 NOT NULL,"
+                    + "  AlbumTitle   STRING(MAX),"
+                    + "  MarketingBudget INT64"
+                    + ") PRIMARY KEY (SingerId, AlbumId),"
+                    + "  INTERLEAVE IN PARENT Singers ON DELETE CASCADE"))
+        .get(10, TimeUnit.MINUTES);
+    databaseId = DatabaseId.of(projectId, instanceId, database);
+  }
+
+  @Before
+  public void insertTestData() {
+    final DatabaseClient client = spanner.getDatabaseClient(databaseId);
+    client.write(
+        Arrays.asList(
+            Mutation.newInsertOrUpdateBuilder("Singers")
+                .set("SingerId")
+                .to(1L)
+                .set("FirstName")
+                .to("Melissa")
+                .set("LastName")
+                .to("Garcia")
+                .build(),
+            Mutation.newInsertOrUpdateBuilder("Albums")
+                .set("SingerId")
+                .to(1L)
+                .set("AlbumId")
+                .to(1L)
+                .set("AlbumTitle")
+                .to("title 1")
+                .set("MarketingBudget")
+                .to(20000L)
+                .build()));
+  }
+
+  @After
+  public void removeTestData() {
+    final DatabaseClient client = spanner.getDatabaseClient(databaseId);
+    client.write(Collections.singletonList(Mutation.delete("Singers", KeySet.all())));
+  }
+
+  @Test
+  public void testAddAndDropDatabaseRole() throws Exception {
+    final String out =
+        SampleRunner.runSample(
+            () ->
+                AddAndDropDatabaseRole.addAndDropDatabaseRole(
+                    projectId, instanceId, databaseId.getDatabase(), "new-parent", "new-child"));
+    assertTrue(out.contains("Created roles new_parent and new_child and granted privileges"));
+    assertTrue(out.contains("Revoked privileges and dropped role new_child"));
+  }
+
+  @Test
+  public void testListDatabaseRoles() throws Exception {
+    final String out =
+        SampleRunner.runSample(
+            () ->
+                ListDatabaseRoles.listDatabaseRoles(
+                    projectId, instanceId, databaseId.getDatabase()));
+    assertTrue(out.contains("new_parent"));
+  }
+
```

Comment:
Can we add integration test for EnableFineGrainedAccess sample as well?

Reply:
No, integration tests for EnableFineGrainedAccess need service account email and hence it was decided to not run integration test for this sample.